src/tasks/shellcheck/task.ts

```diff
@@ -1,12 +1,55 @@
 'use strict';
 
+import path = require('path');
 import taskLib = require('azure-pipelines-task-lib');
+import installer = require('./installer');
+
+const shellcheckExecutable = 'shellcheck';
+
+const handleShellCheckScanFailure = () => {
+    taskLib.setResult(taskLib.TaskResult.Failed, 'ShellCheck scan failed! Check the logs for violation details.');
+};
+
+const getToolRunner = (scriptFiles: string[]) => {
+    let toolRunner = taskLib.tool(shellcheckExecutable);
+    const rootDir = taskLib.cwd();
+    const rootPrefix = path.normalize(`${rootDir}/`);
+
+    scriptFiles.forEach(script => {
+        toolRunner = toolRunner.arg(script.replace(rootPrefix, ''));
```

Comment:
The task lib file finder includes the full directory path to each script file, so using those literally would result in the ShellCheck command looking like the below, with the full directory path prefix on each script name, including in any flagged issues:
```sh
shellcheck /home/vsts/work/1/s/benchmark-performance.sh /home/vsts/work/1/s/dev_setup.sh

In /home/vsts/work/1/s/benchmark-performance.sh line 8:
kill $(jobs -p)
     ^-- SC2046: Quote this to prevent word splitting.
```

That would work, but it's a lot of extra noise for the user. This line here strips out that AZP agent path prefix and displays the file names in a more user-friendly way:

```sh
shellcheck benchmark-performance.sh dev_setup.sh

In benchmark-performance.sh line 8:
kill $(jobs -p)
     ^-- SC2046: Quote this to prevent word splitting.

```